actions/docs-verifier/src/ActionRunner/Program.cs

```diff
@@ -58,24 +41,40 @@ static bool IsInWhatsNewDirectory(PullRequestFile file)
             returnCode++;
         }
     }
+}
+
+return returnCode;
 
-    static bool IsExtensionChangeOnly(string file1, string file2) =>
-        TryStripYmlOrMarkdownExtension(file1, out string strippedFile1) &&
-        TryStripYmlOrMarkdownExtension(file2, out string strippedFile2) &&
-        strippedFile1.Equals(strippedFile2, StringComparison.OrdinalIgnoreCase);
+static bool IsExtensionChangeOnly(string file1, string file2) =>
+    StripExtension(file1).Equals(StripExtension(file2), StringComparison.OrdinalIgnoreCase);
 
-    static bool TryStripYmlOrMarkdownExtension(string file, out string strippedFile)
+static string StripExtension(string file) =>
+    file.Substring(0, file.Length - Path.GetExtension(file).Length);
+
+static bool IsRedirectableFile(PullRequestFile file)
+{
+    string? deletedFileName = file.IsRenamed()
+        ? file.PreviousFileName
+        : (file.IsRemoved() ? file.FileName : null);
+
+    bool isDeletedToc = deletedFileName is not null && deletedFileName.Equals("toc.yml", StringComparison.OrdinalIgnoreCase);
+    // A deleted toc.yml doesn't need redirection.
+    return !isDeletedToc && IsYmlOrMarkdownFile(deletedFileName) && !IsInWhatsNewDirectory(deletedFileName);
+}
+
+static bool IsYmlOrMarkdownFile(string? fileName) => Path.GetExtension(fileName) is ".yml" or ".md";
+
+static bool IsInWhatsNewDirectory(string? fileName)
```

Comment:
Those two functions were operating on `PullRequestFile.FileName`, which is incorrect (but won't probably cause issues in practice). If a file was renamed from `file.badextension` to `file.md`, it will require a redirection before this change.